src/datacatalog/handlers/datasets.py

```diff
@@ -321,21 +319,23 @@ def _add_persistent_links_to(prefix, distributions: T.Iterable[dict]) -> None:
             body='Endpoint does not support * in the If-None-Match header.'
         )
 
-    (docid, selector) = selector.split(':')
-
     try:
         doc, etag = await request.app.hooks.storage_retrieve(
-            app=request.app, docid=docid, etags=etag_if_none_match
+            app=request.app, docid=dataset, etags=etag_if_none_match
         )
-        resource_url = None
-        for distribution in doc.get('dcat:distribution', []):
-            if distribution['dc:identifier'] == selector:
-                resource_url = distribution['dcat:accessURL']
     except KeyError:
         raise web.HTTPNotFound()
-    if doc is None or resource_url is None:
-        return web.Response(status=304, headers={'Etag': etag})
-    raise web.HTTPMovedPermanently(location=resource_url)
+    if doc is None:
+        raise web.HTTPNotModified(headers={'ETag': etag})
+    headers = {'ETag': etag}
+    resource_url = None
+    for distribution in doc.get('dcat:distribution', []):
+        if distribution.get('dc:identifier', None) == distribution:
+            resource_url = distribution.get('dcat:accessURL', None)
+            break
+    if resource_url is None:
+        raise web.HTTPNotFound(headers=headers)
+    raise web.HTTPTemporaryRedirect(location=resource_url, headers=headers)
```

Comment:
Waarom een TemporaryRedirect ipv een MovedPermanently? Bijv bit.ly gebruikt ook 301

Reply:
Bit.ly is bedoeld als URL-shortener, niet als PURL-provider. Bit.ly geeft met die 301 aan dat je, wanneer je deze resource in de toekomst nog een keer wilt opvragen, het beste direct naar de "lange" URL kunt gaan.
In onze use-case willen we juist dat de gebruiker ook in de toekomst de pURL blijft gebruiken, en NIET direct naar de _redirect url_ moet gaan.